src/freenet/client/async/SplitFileInserter.java

```diff
@@ -139,7 +143,9 @@ public void onResume(ClientContext context) throws InsertException, ResumeFailed
                     context.getChkInsertScheduler(realTime).fetchingKeys(), context.persistentFG, 
                     context.persistentFileTracker, context.getPersistentMasterSecret());
             storage.onResume(context);
-            this.sender = new SplitFileInserterSender(this, storage);
+            synchronized(this) {
+                this.sender = new SplitFileInserterSender(this, storage);
```

Comment:
I don't follow why that makes locking necessary. The reads of it are not synchronized, so what does locking this write prevent? Under what circumstances does an insert not yet resumed have methods called from other threads?